src/test/java/com/google/sps/data/friend_graph/GraphCreationTest.java

```diff
@@ -0,0 +1,131 @@
+    userBFriends = new HashSet<>();
+    userCFriends = new HashSet<>();
+    userDFriends = new HashSet<>();
+    userEFriends = new HashSet<>();
+    userA = new UserNode(USER_A_ID, userAFriends);
+    userB = new UserNode(USER_B_ID, userBFriends);
+    userC = new UserNode(USER_C_ID, userCFriends);
+    userD = new UserNode(USER_D_ID, userDFriends);
+    userE = new UserNode(USER_E_ID, userEFriends);
+  }
+
+  @Test
+  public void oneUserGraphTest() {
+    Set<UserNode> oneUserSet = new HashSet<>();
+    oneUserSet.add(userA);
+    UserFriendsGraph oneUserGraph = new UserFriendsGraph(oneUserSet);
+
+    Map<String, Set<String>> friendGraph = oneUserGraph.getFriendGraph();
+    Set<FriendRelationship> friendshipSet = oneUserGraph.getFriendshipEdges();
+
+    Map<String, Set<String>> expectedGraph = new HashMap<>();
+    expectedGraph.put(USER_A_ID, emptyStringSet);
+
+    Assert.assertEquals(friendGraph, expectedGraph);
+    Assert.assertEquals(friendshipSet, emptyFriendshipSet);
+  }
+
+  @Test
+  public void twoUsersNoConnectionsGraphTest() {
+    Set<UserNode> twoUserSet = new HashSet<>();
+    twoUserSet.add(userA);
+    twoUserSet.add(userB);
+    UserFriendsGraph twoUserGraph = new UserFriendsGraph(twoUserSet);
+
+    Map<String, Set<String>> friendGraph = twoUserGraph.getFriendGraph();
+    Set<FriendRelationship> friendshipSet = twoUserGraph.getFriendshipEdges();
+
+    Map<String, Set<String>> expectedGraph = new HashMap<>();
+    expectedGraph.put(USER_A_ID, emptyStringSet);
+    expectedGraph.put(USER_B_ID, emptyStringSet);
+
+    Assert.assertEquals(friendGraph, expectedGraph);
+    Assert.assertEquals(friendshipSet, emptyFriendshipSet);
+  }
+
+  @Test
+  public void threeUsersOneConnectionGraphTest() {
+    userC.addToUserFriends(USER_E_ID);
+    userD.addToUserFriends(USER_E_ID);
+    userE.addToUserFriends(USER_C_ID);
+    userE.addToUserFriends(USER_D_ID);
+    Set<UserNode> threeUsersOneConnectionSet = new HashSet<>();
+    threeUsersOneConnectionSet.add(userC);
+    threeUsersOneConnectionSet.add(userD);
+    threeUsersOneConnectionSet.add(userE);
+    UserFriendsGraph threeUsersOneConnectionGraph = new UserFriendsGraph(threeUsersOneConnectionSet);
+
+    Map<String, Set<String>> friendGraph = threeUsersOneConnectionGraph.getFriendGraph();
+    Set<FriendRelationship> friendshipSet = threeUsersOneConnectionGraph.getFriendshipEdges();
+    
+    Map<String, Set<String>> expectedGraph = new HashMap<>();
+    expectedGraph.put(USER_C_ID, userCFriends);
+    expectedGraph.put(USER_D_ID, userDFriends);
+    expectedGraph.put(USER_E_ID, userEFriends);
+
+    Set<FriendRelationship> expectedFriendshipSet = new HashSet<>();
+    expectedFriendshipSet.add(new FriendRelationship(USER_C_ID, USER_E_ID));
+    expectedFriendshipSet.add(new FriendRelationship(USER_D_ID, USER_E_ID));
+
+    Assert.assertEquals(friendGraph, expectedGraph);
+    Assert.assertEquals(friendshipSet, expectedFriendshipSet);    
+  }
```

Comment:
are there going to be more tests than this later or are these all of them?

Reply:
For now these are all the tests for the initial graph creation. But if any issues come up later on with the graph structure, I'll go back to add more tests in to help fix those issues.